core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreMessageHandler.java

```diff
@@ -104,8 +104,13 @@ private void handleSchemaUpdate(SchemaKey schemaKey, SchemaValue schemaObj) {
   private void tombstoneSchemaKey(SchemaKey schemaKey) {
     tombstoneExecutor.execute(() -> {
           try {
+            while (!schemaRegistry.getKafkaStore().isInitialized()) {
+              Thread.sleep(1000);
```

Comment:
want to use a latch instead?

Reply:
Or are you proposing to use a countdownlatch in the store and await on that?

Reply:
I assumed the later and made a fix since that's more elegant. Let me know what you think about it.